lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -449,6 +452,29 @@ createTransaction w (ApiT wid) body = do
     return $ mkApiTransaction (txId @t tx)
         (fmap Just <$> selection ^. #inputs) (selection ^. #outputs) meta
 
+postExternalTransaction
+    :: forall t k. DefineTx t
+    => WalletLayer (SeqState t) t k
+    -> PostExternalTransactionData
+    -> Handler (ApiTransaction t)
+postExternalTransaction w body = do
+    let externallySignedTx = body ^. #payload
+    tx <- liftHandler $ W.submitExternalTx w externallySignedTx
+    let outs = W.outputs @t tx -- need to check it again
```

Comment:
need to check `outs`, `theAmount` and `ins` once again to make sure they are ok

Reply:
`theAmount` is incorrect - we need inputs and changes (which are missing here). So maybe, I will use sum of outputs minus estimate fee if I can...

Reply:
1. If there's any significant calculation being done, I would move it to a function in Cardano.Wallet.Primitive.Model.
2. Change `theAmount` to `amt`. Or use `amount` and NamedFieldPuns. (The name shadowing warning sort of sucks if it convolutes variable names like this... maybe we could consider disabling it.)
3. To calculate amount properly, I think we need to know which output addresses belong to the sender (the change addresses).... but this we cannot know.